web/src/main/java/org/cbioportal/proxy/ProxyController.java

```diff
@@ -1,38 +1,74 @@
 package org.cbioportal.proxy;
 
-import org.springframework.http.HttpEntity;
-import org.springframework.http.HttpHeaders;
-import org.springframework.http.HttpMethod;
-import org.springframework.http.MediaType;
+import org.cbioportal.web.util.ResettableHttpServletRequestFilter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.*;
 import org.springframework.http.converter.StringHttpMessageConverter;
+import org.springframework.util.StringUtils;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 import org.springframework.web.client.RestTemplate;
 
+import javax.annotation.PostConstruct;
 import javax.servlet.http.HttpServletRequest;
 import java.net.URI;
 import java.net.URISyntaxException;
 import java.nio.charset.StandardCharsets;
 
 @RestController
 public class ProxyController {
+    @Value("${show.oncokb:true}")
+    private Boolean showOncokb;
+    @Value("${oncokb.token:}")
+    private String oncokbToken;
+    @Value("${oncokb.public_api.url:}")
+    private String oncokbPublicApiUrl;
+
+    private Logger LOG = LoggerFactory.getLogger(ProxyController.class);
 
     @RequestMapping("/**")
     public String proxy(@RequestBody(required = false) String body, HttpMethod method, HttpServletRequest request)
         throws URISyntaxException {
+        HttpHeaders httpHeaders = new HttpHeaders();
+        String contentType = request.getHeader("Content-Type");
+        if (contentType != null) {
+            httpHeaders.setContentType(MediaType.valueOf(contentType));
+        }
+        return exchangeData(body, buildUri(request.getPathInfo(), request.getQueryString(), false), method, httpHeaders).getBody();
+    }
 
-        String queryString = request.getQueryString();
-        URI uri = new URI("http:/" + request.getPathInfo() + (queryString == null ? "" : "?" + queryString));
+    @RequestMapping("/oncokb/**")
```

Comment:
Doesn't this provide a route for anybody to make a request for data from oncokb ?  Basically this proxy service allows anybody to make requests to oncokb using the cbioportal server's token. If there is a path like that, doesn't it undermine the effort to add access control to oncokb?

Reply:
It does and it's a loophole. We don't have a better solution at the moment to address the issue. Portal allows cross-domain access, so there is no way to prevent it. We are planning to make some changes to only return the data portal needs, but the way frontend uses oncokb is totally depending on the clients generated through swagger. Which means, if we want to fix the issue, that involves too many changes. In a limited time, we are ok to let the cbioportal has a proxy like this.

Reply:
@sheridancbio the only way to address this issue in long run is to do rate limiting or invisible catcha on the api (or proxy) so that people can't abuse it.  there is no other solution